mlflow/server/js/src/experiment-tracking/pages/experiment-evaluation-datasets/components/ExperimentEvaluationDatasetsIdCell.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import { Tag, Tooltip } from '@databricks/design-system';
+import type { Row } from '@tanstack/react-table';
+import type { EvaluationDataset } from '../types';
+
+export const DatasetIdCell = ({ row }: { row: Row<EvaluationDataset> }) => {
+  const datasetId = row.original.dataset_id;
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const handleClick = () => {
+    navigator.clipboard.writeText(datasetId);
+    setShowTooltip(true);
+    setTimeout(() => {
+      setShowTooltip(false);
+    }, 3000);
+  };
+
+  return (
+    <Tooltip
+      content={showTooltip ? 'Copied!' : 'Click to copy'}
```

Comment:
nit: i18n (we don't i18n in OSS but it's a best practice so we don't have i18n issues on backport) also probably unnecessary to have this be a ternary since if the tooltip is not shown they won't see the message anyway

```suggestion
      content={
        <FormattedMessage defaultMessage="Copied!" description="Tooltip text shown when the user copies the tag text" />
      }
```